api/v1/lib/scheduler/events/events.go

```diff
@@ -133,53 +159,71 @@ func DefaultHandler(eh Handler) Option {
 	}
 }
 
+// AckError wraps a caller-generated error and tracks the call that failed.
+type AckError struct {
+	Ack   *scheduler.Call
+	Cause error
+}
+
+func (err *AckError) Error() string { return err.Cause.Error() }
+
 // AcknowledgeUpdates generates a Handler that sends an Acknowledge call to Mesos for every
-// UPDATE event that's received.
-func AcknowledgeUpdates(callerGetter func() calls.Caller) Handler {
+// UPDATE event that's received (that requests an ACK).
+func AcknowledgeUpdates(callerLookup func() calls.Caller) Handler {
```

Comment:
redundant w/ the rule version. can we refactor this leverage the Rule implementation?